File: 844. Backspace String Compare/python/main.py
class Solution:

    def buildString(self, s: str) -> str:
        stack = []
        for char in s:
            if char == '#':
                stack.pop()
            else:
                stack.append(char)
        return ''.join(stack)

    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.buildString(s) == self.buildString(t)


if __name__ == "__main__":
    s = "ab#c"
    t = "ad#c"
    print(Solution().backspaceCompare(s, t))

File: 844. Backspace String Compare/javascript/main.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

var buildString = function (s) {
    let stack = [];
    for (let i = 0; i < s.length; i++) {
        if (s[i] === '#') {
            stack.pop();
        } else {
            stack.push(s[i]);
        }
    }
    return stack.join('');
}
var backspaceCompare = function(s, t) {
    return buildString(s) === buildString(t);
};

console.log(backspaceCompare("ab#c", "ad#c"));
